src/docstrings.jl

```diff
@@ -0,0 +1,154 @@
+@doc """
```

Comment:
these should probably just be added directly to the function definitions, but this is fine.